cmd/lotus/daemon.go

```diff
@@ -574,3 +600,88 @@ func ImportChain(ctx context.Context, r repo.Repo, fname string, snapshot bool)
 
 	return nil
 }
+
+func slashConsensus(a lapi.FullNode, from string, extra string) error {
+	ctx := context.Background()
+
+	blocks, err := a.SyncIncomingBlocks(ctx)
+	if err != nil {
+		return err
```

Comment:
Please chain this and all other errors for clarity (that is, `return xerrors.Errorf("sync incoming blocks failed: %w", err)`.